packages/xo-web/src/common/xo/index.js

```diff
@@ -2890,6 +2893,49 @@ export const deleteSshKeys = keys =>
     })
   }, noop)
 
+export const addAuthToken = async () => {
+  const { description, expiration } = await confirm({
+    body: <NewAuthTokenModal />,
+    icon: 'user',
+    title: _('newAuthTokenModalTitle'),
+  })
+  const expires = new Date(expiration).setUTCHours(23, 59, 59)
+  return _call('token.create', {
+    description,
+    expiresIn: Number.isNaN(expires) ? undefined : expires - new Date().getTime(),
```

Comment:
should we test if the expiration date is in the future ? what is the api comportment in this case

Reply:
Sure, I'll add that, the method will error if `expires` is negative.

Reply:
Done: https://github.com/vatesfr/xen-orchestra/commit/4f50f902138948aee81a4ec0d505778e9f3a0558